staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -214,6 +185,7 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 	}
 	defaultVersionedObject := indirectArbitraryPointer(versionedPtr)
 	kind := fqKindToRegister.Kind
+	clusterScoped := a.group.RootScopedKinds.Has(kind)
```

Comment:
Does this assume the kind is in the installer api group? Does that hold for cross-group subresources?

Reply:
> a.group.RootScopedKinds.Has(kind)

I'll trace again to be sure, but I was fairly certain that the choice of cluster and non-cluster scoped was done against the parent kind referenced here similar to the mapping it replaced.

Reply:
> Does this assume the kind is in the installer api group? Does that hold for cross-group subresources?

Alright, I see it.  If you're a `GroupVersionKindProvider` you should be forced to provide scope information to back the kind since it can be external.  I'll add type assertions, then swizzle in the interface.  In the non-override case, they are guaranteed in the same group.